### ISSUE TITLE
Option to docker datasource to ignore new tags if manifest is the same

### ISSUE THREAD

**Issue by ns-ggeorgiev** [renovatebot/renovate#15881]
### What would you like Renovate to be able to do?

When looking for updates currently renovate scans docker registry for "new" tags. If they are there, it will suggest update for it. On other side - from docker registry perspective tags are considered volatile, since they simply refer to shared files. Obviously if a new tag triggers a big sequence of updates with expensive testing etc. they are not so volatile anymore.

I have a scenario where a single repo is responsible for creating a big number of images and they are released frequently. Though only small set of these images is actually updated from release to release.

Tagging all the images on release makes sense since if a developer needs to pick an image they should be able to pick the tag that was announced as part of the release.

On the other hand components that already have commited to existing tag should not need to update to the new tag if the image is exactly the same (except for the tag).

Long story short, I am looking for renovate to compare the manifests for the current and the latest tag, and if they match to ignore the new tag.

### If you have any ideas on how this should be implemented, please tell us here.

All we need is the docker datasource to add an extra check after the new tag is determined and compare the manifests for the current and the new tag. If the same, do not suggest the new tag.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
Can you start with a reproduction? Including showing how you reference it. e.g. you reference using both tag and digest, or tag only?

**Comment by viceice** [comment]
i don't like that idea, as we need to fetch the digest for every tag, so renovate will be rate limited very fast (at least for free users on docker hub)

btw: that's the same reason why we don't try to get the release date of every tag.

**Comment by ns-ggeorgiev** [comment]
@viceice 

> fetch the digest for every tag

no, the manifest only for the current and the new tag will need to be fetched

**Comment by ns-ggeorgiev** [comment]
> Can you start with a reproduction? Including showing how you reference it. e.g. you reference using both tag and digest, or tag only?

what difference that would make. At the end you have reference for current version and reference for a new version. If these two are technically the same (same manifest) then do not make the suggestion.

**Comment by viceice** [comment]
still to many, as there can be multiple updates(major, minor, patch) at same time.

that will cause issues on terraform repos where a lot docker images are used

**Comment by ns-ggeorgiev** [comment]
it could be an option

**Comment by rarkins** [comment]
Maybe we can have an option that goes something like this: "Skip an update if the digest is the same, even currentValue/newValue differ". But please start with a reproduction

**Comment by ns-ggeorgiev** [comment]
> still to many, as there can be multiple updates(major, minor, patch) at same time.
> 
> that will cause issues on terraform repos where a lot docker images are used

I am not sure I get the concern. What I am assuming is that such checks for the same manifest will not be more than actual PRs created. Customer usually throttle the number of PRs they get in a single session. Is there something that I am not accounting for.

Maybe one thing that could be considered for optimization is, how to prevent this check to happen again and again from scan to scan.

**Comment by ns-ggeorgiev** [comment]
> 



> Maybe we can have an option that goes something like this: "Skip an update if the digest is the same, even currentValue/newValue differ". But please start with a reproduction

I have a lot of things to do, and "maybe" is not a good enough for me to spend effort on this. Lets first agree if you will accept the change or not. I think the goal is clear without reproduction.

**Comment by viceice** [comment]
yes, you need to know a lot of renovate internal processing.

i expect that such a feature isn't easy to implement
